packages/nodes-base/nodes/Clockify/ClockifyWriter.node.ts

```diff
@@ -0,0 +1,529 @@
+
+import {IClientDto, IWorkspaceDto} from "./WorkpaceInterfaces";
+import {IUserDto} from "./UserDtos";
+import {runInThisContext} from "vm";
+import {IProjectDto, ITaskDto} from "./ProjectInterfaces";
+import {ITagDto} from "./CommonDtos";
+import {ITimeEntryDto, ITimeEntryRequest} from "./TimeEntryInterfaces";
+import {stringify} from "querystring";
+import {callbackify} from "util";
+import { goalKeyResultFields } from '../ClickUp/GoalKeyResultDescription';
+
+export class ClockifyWriter implements INodeType {
+	description: INodeTypeDescription = {
+		displayName: 'Clockify',
+		name: 'clockifyWriter',
+		icon: 'file:clockify.png',
+		group: ['transform'],
+		version: 1,
+		description: 'Access data on Clockify',
+		defaults: {
+			name: 'Clockify',
+			color: '#772244',
+		},
+		inputs: ['main'],
+		outputs: ['main'],
+		credentials: [
+			{
+				name: 'clockifyApi',
+				required: true,
+			}
+		],
+		properties: [
+			{
+				displayName: 'Operation',
+				name: 'operation',
+				type: 'options',
+				options: [
+					{
+						name: 'Create',
+						value: 'create'
+					},
+					{
+						name: 'Update',
+						value: 'update'
+					},
+					{
+						name: 'Delete',
+						value: 'delete'
+					},
+				],
+				default: 'create',
+				description: 'The operation you wish to carry out.',
+			},
+			{
+				displayName: 'Resource',
+				name: 'resource',
+				type: 'options',
+				options: [
+					{
+						name: 'Project',
+						value: 'project'
+					},
+					{
+						name: 'Tag',
+						value: 'tag'
+					},
+					{
+						name: 'Time Entry',
+						value: 'timeEntry'
+					},
+				],
+				default: 'project',
+				description: 'The resource to operate on.',
+			},
+			{
+				displayName: 'Workspace',
+				name: 'workspaceId',
+				type: 'options',
+				typeOptions: {
+					loadOptionsMethod: 'listWorkspaces',
+				},
+				required: true,
+				default: [],
+			},
+			{
+				displayName: 'User',
+				name: 'userId',
+				type: 'options',
+				typeOptions: {
+					loadOptionsDependsOn: ['workspaceId'],
+					loadOptionsMethod: 'loadUsersForWorkspace',
+				},
+				required: true,
+				default: [],
+				displayOptions: {
+					show: {
+						resource: [
+							'timeEntry',
+						],
+					},
+				},
+			},
+			{
+				displayName: 'Client',
+				name: 'clientId',
+				type: 'options',
+				typeOptions: {
+					loadOptionsDependsOn: ['workspaceId'],
+					loadOptionsMethod: 'loadClientsForWorkspace',
+				},
+				required: true,
+				default: [],
+			},
+			{
+				displayName: 'Project',
+				name: 'project',
+				type: 'options',
+				typeOptions: {
+					loadOptionsDependsOn: ['workspaceId'],
+					loadOptionsMethod: 'loadProjectsForWorkspace',
+				},
+				required: true,
+				default: [],
+				description: 'Project to associate with, leaving blank will use the project associated with the task.',
+				displayOptions: {
+					hide: {
+						resource: [
+							'project',
+						],
+					},
+				}
+			},
+			{
+				displayName: 'Billable?',
```

Comment:
This renders weird with double punctuation.
```suggestion
				displayName: 'Billable',
```